### PR TITLE
Add forms for collecting data on posts and comments and functionality for likes

### PR DESCRIPTION
**Pull request by j-tee**
# In This Milestone
-  Create a method called `current_user` in ApplicationController to make current user data available to all controllers.
    -  It returns the first user from the database.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` 
    - Create a comment on behalf of the `current_user` 
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    end
+    redirect_to user_post_path(user_id: @post.user_id, id: @post.id)
+  end
+
+  def update; end
+
+  def destroy; end
+
+  private
+
```
Comment (sja-thedude) [inline app/controllers/comments_controller.rb:20]: - [ ] Kindly fix the error which s displayed when I try creating comments on rails c.

Screenshot 2023-01-27 at 2 25 26 PM

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @j-tee, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters passing
✔️ Professional Readme
✔️ Correct Gitflow

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (j-tee) [reply app/controllers/comments_controller.rb:20]: > * [ ]  Kindly fix the error which s displayed when I try creating comments on rails c.
> 
> Screenshot 2023-01-27 at 2 25 26 PM

image
image

Comment (j-tee) [reply app/controllers/comments_controller.rb:20]: As you can see from my screenshots I can add comments from my end sucessfully.

Comment (j-tee) [reply app/controllers/comments_controller.rb:20]: If you dont mind I would like to request a second opinion

File: app/views/posts/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  <div class="form-group">
+    <%= form.label :comments_counter %>
+    <%= form.number_field :comments_counter, class: 'form-control', value: @post.comments_counter %>
+  </div>
+  <div class="form-group">
+    <%= form.label :likes_counter %>
+    <%= form.number_field :likes_counter, class: 'form-control', value: @post.likes_counter %>
+  </div>
+  <%= form.submit "Create Post", class: "btn btn-primary" %>
+<% end %>
```
Comment (elisha2kyakpo1) [inline app/views/posts/new.html.erb]: - [ ] The likes counter and comments counter shouldn't be created by the user. The counters will count when a comment or like is created.

counter

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @j-tee, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] The error which was mentioned by the previous reviewer is coming from the seed file. It can be good if you fix it because if users run a seed file, they are likely to have a problem.

seed

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/new.html.erb (hunk 2 above)
Comment (j-tee) [reply app/views/posts/new.html.erb]: Oh yeah, forgot about it

**Review by sumairq — APPROVED** [review]
### Hi @j-tee , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**